lib/ansible/cli/playbook.py

```diff
@@ -89,27 +90,26 @@ def run(self):
 
         # initial error check, to make sure all specified playbooks are accessible
         # before we start running anything through the playbook executor
+        pb_dirs = []
         for playbook in self.args:
             if not os.path.exists(playbook):
                 raise AnsibleError("the playbook: %s could not be found" % playbook)
             if not (os.path.isfile(playbook) or stat.S_ISFIFO(os.stat(playbook).st_mode)):
                 raise AnsibleError("the playbook: %s does not appear to be a file" % playbook)
+            pb_dirs.append(os.path.dirname(playbook))
```

Comment:
This line is the reason that `The module helloworld_linux was not found in configured module paths` is happening.

`os.path.dirname(playbook)` on something like `test_binary_modules.yml` just becomes `''`.

```suggestion
            pb_dirs.append(os.path.dirname(os.path.abspath(playbook)))
```

Otherwise, you can handle using `abspath` in `add_all_plugin_dirs`